imports/plugins/core/email/client/containers/emailConfig.js

```diff
@@ -1,38 +1,75 @@
+import React, { Component } from "react";
 import { useDeps } from "react-simple-di";
 import getServiceConfig from "nodemailer-wellknown";
 import { Meteor } from "meteor/meteor";
+import PropTypes from "prop-types";
 import { Reaction } from "/client/api";
 import { Loading } from "/imports/plugins/core/ui/client/components";
 import actions from "../actions";
 import EmailConfig from "../components/emailConfig";
 import { composeWithTracker, merge } from "/lib/api/compose";
 
-const composer = ({}, onData) => {
-  if (Meteor.subscribe("Packages").ready()) {
-    const shopSettings = Reaction.getShopSettings();
-    const settings = shopSettings.mail || {};
+class EmailConfigContainer extends Component {
 
-    if (settings.service && settings.service !== "custom") {
-      const config = getServiceConfig(settings.service);
+  constructor(props) {
+    super(props);
 
-      // show localhost for test providers like Maildev that have no host
-      settings.host = config.host || "localhost";
-      settings.port = config.port;
-    }
+    this.state = {
+      status: null,
+      error: null
+    };
+  }
 
+  componentWillMount() {
+    const { settings } = this.props;
     const { service, host, port, user, password } = settings;
 
-    // if all settings exist, check if they work
     if (service && host && port && user && password) {
       Meteor.call("email/verifySettings", (error) => {
         if (error) {
-          return onData(null, { settings, status: "error", error: error.reason });
+          this.setState({ status: "error" });
```

Comment:
Yeah, it should be but that's not what this ticket is about so that's why I already approved it.